feat: added new rule "attr-space-between"

actengage opened the pull request:
Consider the following HTML:

```html
<div id="bar"class="foo"></div>
```

The following errors are what you get today:

```
Special characters must be escaped : [ &lt; ].
Special characters must be escaped : [ &gt; ].
Tag must be paired, no start tag: [ &lt;/div&gt; ]
```

The `attr-space-between` rule ensures HTML tags with attributes must have spaces between them. Without this change, a tag with attributes missing the required space results in the parser interpreting the tag as text, which produces the misleading errors above from unrelated rules. With this change the error becomes:

```
Attribute [ class ] must be separated with a space.
```

**Proposed changes:**

1. Loosen `regTag` in `htmlparser.ts` (`\s+` → `\s*` before attributes) so these tags still parse instead of falling through as text.
2. Add the new `attr-space-between` rule to report the missing space directly.

Rebased onto current `main` (2026-07): rule/test/docs now follow the current conventions (`website/src/content/docs/rules/*.mdx`, `Ruleset` type entry, bracket-style error message). All 376 tests pass, lint/prettier clean.

coliff commented:
Thanks for the PR! It looks good to me at a glance and could be an helpful addition to HTMLHint.

We need @thedaviddias to review it.

actengage commented:
I didn't see this reply before. Thanks for looking into this. I'd love to get this approved in a new release so I don't have to use my fork in production.

coliff commented:
> I didn't see this reply before. Thanks for looking into this. I'd love to get this approved in a new release so I don't have to use my fork in production.

No problem. I'd like to see this added too but we need @thedaviddias to review it as he is the project owner but he hasn't been active here for a couple of years.

coliff commented:
There are conflicts in the branch now so moving this to Draft status.

Commit:
feat: added new rule "attr-space-between”

Without a space between attributes, the parser interprets the tag as text, producing misleading spec-char-escape/tag-pair errors. Loosen the tag regex so such tags still parse, and report the missing space directly via the new attr-space-between rule.

actengage commented:
Hello! I'm really glad to see this project has been updated since I last looked at it. I was worried it was dead for a while. I haven't updated my production code that uses this project in over year and I was curious if this PR was still here. I fixed the merge conflicts and rebased against the most recent main branch. Let me know if there is anything more I can do! Thank you very much.
